src/index.js

```diff
@@ -115,7 +115,14 @@ class KadDHT {
    */
   start (callback) {
     this._running = true
-    this.network.start(callback)
+    this.network.start((err) => {
+      if (err) {
+        return callback(err)
+      }
+      this.randomWalk.start()
```

Comment:
This should be started just like any other Peer Discovery Service, that means, start it on js-libp2p instead.

Reply:
I will change the PR and create a new one for `js-libp2p`! Thanks